pkl-cli/pkl-cli.gradle.kts

```diff
@@ -121,33 +130,95 @@ val testJavaExecutable by
         (configurations.testRuntimeClasspath.get() - configurations.runtimeClasspath.get())
   }
 
-tasks.check { dependsOn(testJavaExecutable) }
+// Setup `testJavaExecutable` tasks for multi-JDK testing.
+val testJavaExecutableOnOtherJdks =
+  if (buildInfo.multiJdkTesting) {
+    buildInfo.multiJdkTestingWith(testJavaExecutable)
+  } else {
+    emptyList()
+  }
```

Comment:
Just like multi-JDK tests, `testJavaExecutableOnJdk*` and friends now only run as dependencies of `check` iff `(we are in ci) OR (the local dev has activated multi-JDK testing)`.